corehq/apps/reports/views.py

```diff
@@ -983,16 +983,24 @@ def _sort_key(config_choice):
     def scheduled_report_form(self):
         web_users = WebUser.view('users/web_users_by_domain', reduce=False,
                                key=self.domain, include_docs=True).all()
-        web_user_emails = [u.get_email() for u in web_users]
         initial = self.report_notification.to_json()
-        initial['recipient_emails'] = ', '.join(initial['recipient_emails'])
         kwargs = {'initial': initial}
-        args = ((self.request.POST, ) if self.request.method == "POST" else ())
+        if self.request.method == "POST":
+            args = (self.request.POST, )
+            selected_emails = self.request.POST.getlist('recipient_emails', {})
+        else:
+            args = ()
+            selected_emails = kwargs.get('initial', {}).get('recipient_emails', [])
+
+        web_user_emails = [u.get_email() for u in web_users]
```

Comment:
I'd prefer to put this logic - the "choices must include any selected values" logic - in `MultiCharField` but wasn't able to find a way to access the field's value inside of `MultiCharField`.